src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -265,22 +265,27 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
     )
   }
 
-  def cloneWorkspace(sourceWorkspaceName: WorkspaceName, destWorkspaceName: WorkspaceName): Future[PerRequestMessage] =
-    dataSource.inFutureTransaction(readLocks=Set(sourceWorkspaceName), writeLocks=Set(destWorkspaceName)) { txn =>
+  def cloneWorkspace(sourceWorkspaceName: WorkspaceName, destWorkspaceRequest: WorkspaceRequest): Future[PerRequestMessage] =
+    dataSource.inFutureTransaction(readLocks=Set(sourceWorkspaceName), writeLocks=Set(destWorkspaceRequest.toWorkspaceName)) { txn =>
       withWorkspaceContextAndPermissions(sourceWorkspaceName,WorkspaceAccessLevels.Read,txn) { sourceWorkspaceContext =>
-        withNewWorkspaceContext(
-          WorkspaceRequest(
-            namespace = destWorkspaceName.namespace,
-            name = destWorkspaceName.name,
-            realm = sourceWorkspaceContext.workspace.realm,
-            attributes = sourceWorkspaceContext.workspace.attributes),txn) { destWorkspaceContext =>
+
+        // add to or replace current attributes, on an individual basis
+        val newAttrs = sourceWorkspaceContext.workspace.attributes ++ destWorkspaceRequest.attributes
+
+        // if the source has a realm, the dest must also have that realm.  Otherwise, the caller chooses
+        val newRealm = sourceWorkspaceContext.workspace.realm match {
+          case Some(realm) => Some(realm)
```

Comment:
should we throw an error if both the source and destination define a realm that are different?